add language for non-compliant JSON-LD processors (#563)
openid/OpenID4VP: openid-4-verifiable-presentations-1_0.md

The following is a W3C Verifiable Credentials specific parameter in the `meta` parameter in a Credential Query as defined in (#credential_query):

`type_values`:

> : REQUIRED. An array of string arrays that specifies the fully expanded types (IRIs) after the `@context` was applied that the Verifier accepts to be presented in the Presentation. If a type value was not defined by any `@context`, its value stays the same. This works fine in practice, even if it's not strictly JSON-LD compliant. Implementations can use alternative mechanisms that produce the same outcome as applying the `@context`, provided that the result is equivalent. Each of the top-level arrays specifies one alternative to match the `type` values of the Verifiable Credential against. Each inner array specifies a set of fully expanded types that MUST be present in the `type` property of the Verifiable Credential, regardless of order or the presence of additional types. 

Reviewer: Maybe I am missing something but the later part feels redundant?
```suggestion
: REQUIRED. An array of string arrays that specifies the fully expanded types (IRIs) after the `@context` was applied that the Verifier accepts to be presented in the Presentation. If a type value was not defined by any `@context`, its value stays the same. This works fine in practice, even if it's not strictly JSON-LD compliant. Implementations can use alternative mechanisms that produce the same outcome as applying the `@context`. Each of the top-level arrays specifies one alternative to match the `type` values of the Verifiable Credential against. Each inner array specifies a set of fully expanded types that MUST be present in the `type` property of the Verifiable Credential, regardless of order or the presence of additional types. 
```

Reviewer 2: ```suggestion
: REQUIRED. An array of string arrays that specifies the fully expanded types (IRIs) after the `@context` was applied that the Verifier accepts to be presented in the Presentation. If a type value was not defined by any `@context`, its value stays the same as it is effectively a relative IRI. Implementations can use alternative mechanisms that produce the same outcome as applying the `@context`, provided that the result is equivalent. Each of the top-level arrays specifies one alternative to match the `type` values of the Verifiable Credential against. Each inner array specifies a set of fully expanded types that MUST be present in the `type` property of the Verifiable Credential, regardless of order or the presence of additional types. 
```

> : REQUIRED. An array of string arrays that specifies the fully expanded types (IRIs) after the `@context` was applied that the Verifier accepts to be presented in the Presentation. If a type value was not defined by any `@context`, its value stays the same. This works fine in practice, even if it's not strictly JSON-LD compliant. Implementations can use alternative mechanisms that produce the same outcome as applying the `@context`, provided that the result is equivalent. Each of the top-level arrays specifies one alternative to match the `type` values of the Verifiable Credential against. Each inner array specifies a set of fully expanded types that MUST be present in the `type` property of the Verifiable Credential, regardless of order or the presence of additional types. 

Reviewer 3: few clarification questions:

> If a type value was not defined by any `@context`, its value stays the same. 

what value?

> This works fine in practice, even if it's not strictly JSON-LD compliant. 

i would just say `This works in practice` to make it a bit more formal and avoid talking about JSON-LD compliance..?

> Each of the top-level arrays specifies one alternative to match the `type` values of the Verifiable Credential against. 

which top level arrays..?

Reviewer 2: > what value?

I believe what @awoie is referring to is the original type value in its non-expanded JSON-LD form

> i would just say This works in practice to make it a bit more formal and avoid talking about JSON-LD compliance..?

I would go one step further and not even say "works in practise" the fact we have defined this in the spec means implicitly it works.

> : REQUIRED. An array of string arrays specifying the fully expanded types (IRIs) that the Verifier accepts in a Presentation, after applying the `@context` to the Verifiable Credential. If a `type` value in a Verifiable Credential is not defined in any `@context`, it remains unchanged, i.e., remains a relative IRI after JSON-LD processing. For this reason, JSON-LD processing MAY be skipped in such cases and the relative IRI is considered to be the fully expanded type, as applying the `@context` would not alter the value. Implementations MAY use alternative mechanisms to obtain the fully expanded types, as long as the results are equivalent to those produced by JSON-LD processing. Each of the top-level arrays specifies one alternative to match the fully expanded `type` values of the Verifiable Credential against. Each inner array specifies a set of fully expanded types that MUST be present in the fully expanded types in the `type` property of the Verifiable Credential, regardless of order or the presence of additional types. 

Reviewer 3: ```suggestion
: REQUIRED. An array of strings, where each string specifies a fully expanded type (IRI) that the Verifier accepts in a Presentation, after applying the `@context` to the Verifiable Credential. If a `type` value in a Verifiable Credential is not defined in any `@context`, it remains unchanged, i.e., remains a relative IRI after JSON-LD processing. For this reason, JSON-LD processing MAY be skipped in such cases and the relative IRI is considered to be the fully expanded type, as applying the `@context` would not alter the value. Implementations MAY use alternative mechanisms to obtain the fully expanded types, as long as the results are equivalent to those produced by JSON-LD processing. Each of the top-level arrays specifies one alternative to match the fully expanded `type` values of the Verifiable Credential against. Each inner array specifies a set of fully expanded types that MUST be present in the fully expanded types in the `type` property of the Verifiable Credential, regardless of order or the presence of additional types. 
```

}
```

The following is another non-normative example of a W3C Verifiable Credential that would match the `type_values` DCQL query above (other claims omitted for readability):

```json
{
  "@context": [
    "https://www.w3.org/2018/credentials/v1"
  ],

>   "type": ["VerifiableCredential", "IdentityCredential"]

Reviewer 4: In this case, the "IdentityCredential" is silently dropped by JSON-LD Processors, such as the one used on json-ld playground:

https://json-ld.org/playground/

```
_:b0 <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <https://www.w3.org/2018/credentials#VerifiableCredential> .
```

Try pasting the JSON-LD above in, and note the errors and silent dropping when changing tabs.
